src/runner.rs

```diff
@@ -53,39 +53,40 @@ impl LlmRunner {
 
         // TODO: To make type to cast conditional to support various of protocols
         let result = provider
-            .execute_request::<OpenAIResponse>(
+            .execute_request::<CreateCompletionResponse>(
                 request,
                 Arc::clone(&sender),
                 Arc::clone(&cancel_flag),
                 assistant_settings.stream,
             )
             .await;
 
-        if let Some(tool_calls) = result
+        if let Some(tool_calls_text) = result
             .as_ref()
             .ok()
             .and_then(|r| {
                 r.choices
                     .first()?
-                    .message
-                    .tool_calls
+                    .text
                     .clone()
             })
         {
             if let Ok(ref message) = result {
+                let assistant_message = AssistantMessage {
+                    role: Roles::Assistant,
+                    content: Some(tool_calls_text.clone()),
+                    tool_calls: None,
+                };
                 cacher
                     .lock()
                     .await
-                    .write_entry(&CacheEntry::from(
-                        message.clone().choices[0]
-                            .message
-                            .clone(),
-                    ))
+                    .write_entry(&CacheEntry::from(assistant_message))
                     .ok();
             }
 
+            // TODO: Parse tool_calls from tool_calls_text if needed
```

Comment:
I'm afraid this one in particular is blocker to be merged. It's actually one of the main feature of plugin -- robust agentic capabilities.

So I need this bit working in full scale.